homeassistant/components/tibber/sensor.py

```diff
@@ -328,14 +403,93 @@ async def async_setup_entry(
     async_add_entities(entities, True)
 
 
-class TibberSensor(SensorEntity):
-    """Representation of a generic Tibber sensor."""
+async def _async_setup_data_api_sensors(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up sensors backed by the Tibber Data API."""
+
+    runtime = hass.data.get(DOMAIN)
+    if runtime is None:
+        raise PlatformNotReady("Tibber runtime is not ready")
+    coordinator = TibberDataAPICoordinator(hass, entry, runtime)
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
It's better practice to create and refresh the coordinator in the init module `async_setup_entry` since the coordinator shouldn't really be tied to a specific platform.